support-frontend/assets/helpers/internationalisation/helpers.ts

```diff
@@ -0,0 +1,411 @@
+import * as cookie from 'helpers/storage/cookie';
+import type { Option } from 'helpers/types/option';
+import { getQueryParameter } from 'helpers/urls/url';
+import type {
+	AuState,
+	CaState,
+	IsoCountry,
+	StateProvince,
+	UsState,
+} from './country';
+import {
+	auStates,
+	caStates,
+	countries,
+	isoCountries,
+	isoCountrySet,
+	usStates,
+} from './country';
+import {
+	AUDCountries,
+	Canada,
+	countryGroups,
+	EURCountries,
+	GBPCountries,
+	International,
+	NZDCountries,
+	UnitedStates,
+} from './countryGroup';
+import type {
+	CountryGroup,
+	CountryGroupId,
+	CountryGroupName,
+} from './countryGroup';
+
+type TargetCountryGroups =
+	| typeof International
+	| typeof EURCountries
+	| typeof NZDCountries
+	| typeof GBPCountries
+	| typeof AUDCountries;
+
+class CountryHelper {
+	static stateProvinceFromMap(
+		l: string,
+		states: Record<string, string>,
+	): StateProvince | null | undefined {
+		const s = l.toUpperCase();
+		return states[s]
+			? s
+			: Object.keys(states).find(
+					(key) =>
+						states[key].toUpperCase() === s ||
+						(s.length === 3 && s.startsWith(key)),
+			  );
+	}
+
+	static usStateFromString(s: string): Option<UsState> {
```

Comment:
I'd avoid using the `Option` type in new code; it's unfortunately a bit tricky to completely remove from the codebase, but it's clearer to just use `Type | null`- this is all our custom `Option` type actually is, but the name is quite deceptive as that's not at all what an `Option` is in languages that really have them like Scala or Rust.